apps/total-typescript/src/video/exercise-overlay.tsx

```diff
@@ -125,20 +128,84 @@ const ExerciseOverlay = () => {
     type: lesson._type,
   })
   const {github} = module
+  const {isSafari, isFirefox} = useDeviceDetect()
+  const isStackblitzCompatibleBrowser = !(isSafari || isFirefox)
+
+  const {exerciseGitHubUrl} = getExerciseGitHubUrl({stackblitz, module})
 
   return status !== 'loading' ? (
     <div className=" bg-black/30 ">
       {stackblitz ? (
         <>
           <div className="flex w-full items-center justify-between p-3 pl-5 font-medium sm:text-lg">
-            <div>Now it's your turn! Try solving this exercise.</div>
+            <div className="flex flex-col">
+              <div>Now it's your turn! Try solving this exercise.</div>
+            </div>
             <div className="flex justify-center gap-2">
               <Actions />
             </div>
           </div>
           <div className="relative hidden h-[500px] w-full sm:block xl:h-[750px]">
             <StackBlitzIframe exercise={lesson} module={module} />
           </div>
+          {!isStackblitzCompatibleBrowser && (
+            <div className="mx-2 mt-2 hidden rounded-md bg-gray-800 px-4 py-3 text-base sm:block">
+              <p className="pb-1 font-semibold">
+                <ExclamationIcon className="inline-block h-5 w-5 text-cyan-300" />{' '}
+                StackBlitz is poorly supported outside of Chrome.
+              </p>
+              <p>
+                {isFirefox && (
+                  <>
+                    {
+                      'Please use Chromium-based browser to work on this exercise. Or '
+                    }
+                    <a
+                      href={`https://gitpod.io#${exerciseGitHubUrl}`}
```

Comment:
I changed both of these to link to gitpod instead of stackblitz or into github since it 1. gives them an immediate env 2. also connects to VS Code directly

Reply:
nice!